[WIP] Separate components example

AndrewZamora opened the pull request:
Moved Login component to `src` directory and renamed `component` directory to `example` for #3.
Should I move `App.js`, `App.css`, `App.test.js`, `index.css`, `index.js` and `registerServiceWorker.js` to the `example` directory? 
![image](https://user-images.githubusercontent.com/31365406/46808814-21cbbc00-cda8-11e8-97f0-2eba85d4543f.png)

martinheidegger commented:
The reason for this exercise is that the example code is separated from the component. Once the component is separate from the example: you can publish the component as is and the example can work as documentation. For that to work well, its better to have the component entirely separated: 

```
- src
  - Login.js
- example
  - src
    - App.js
      ...
```

AndrewZamora commented:
I've have created another `src` folder and moved rest example code into it. I have also moved the example directory in into the root directory. 

In terms of further development will I need create an 'index.js' and so on to get `create-react-app` to run? Or am I getting ahead of myself?

AndrewZamora commented:
@martinheidegger I tried to export the login component from the root directory's `src` into the `example` directory and ran into this:
![image](https://user-images.githubusercontent.com/31365406/46904529-3df26900-cf20-11e8-853f-a9a6e6d6adb2.png)
Then I took a look at the `create-react-app` documentation and saw this under the `you might want to try something else` section:
![image](https://user-images.githubusercontent.com/31365406/46904569-5a42d580-cf21-11e8-9924-5b3b9058b32c.png)

Should I run `npm eject` to change to webpack settings?

AndrewZamora commented:
@martinheidegger I added the dependencies to the `package.json` file. I'm assuming `"react-login-form": "latest"` dependency. Won't work till I publish it on NPM.

martinheidegger commented on example/package.json:
```diff
@@ -0,0 +1,17 @@
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "react": "^16.5.2",
```
You can add your library as a dependency here!

`"react-login-page": "https://github.com/AndrewZamora"`

Some notes here:

1.) This changes the imports below `/example` to from `../../src/Login` to `react-login-page`
2.) It would be good to make sure that the "name" in the `./package.json` is maybe better, like: `react-login-form` or `@AndrewZamora/react-login` (`@AndrewZamora` being the npm username!)
3.) You would probably need to have a compiled version of the `src` folder in the repository for this to work.
4.) As-is the example page wouldn't work, you need to trick a bit to make the local dev environment work properly.

martinheidegger commented:
This is a massive update, how do you think about it?

AndrewZamora commented:
Well I feel really comfortable going through the `node_modules` directory now 😂. I understand now that I can install code from a git repository through NPM and that the code needs to be compiled into a build directory for it to work. A lot of this process has been demystified for me. However, I think I could have broken this pull request into two. First, just setting up the file structure for the `src` and `example` directories and maybe adding the build directory. Second, adding the babel configuration and the build scripts. In the future think I need to focus more on the objective at hand.
